app/src/main/java/com/example/neighborly/RequestActivity.java

```diff
@@ -215,4 +223,49 @@ public void onStop() {
     int getImage() {
         return 0;
     }
+
+    private void showResolvedPopup() {
+        popupRequestDialog = new Dialog(this);
+        popupRequestDialog.setContentView(R.layout.popup_resolved_request);
+
+        Button closeButton = popupRequestDialog.findViewById(R.id.exit);
+        final Button sendButton = popupRequestDialog.findViewById(R.id.send);
+
+        sendButton.setVisibility(View.INVISIBLE);
+
+        closeButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                popupRequestDialog.dismiss();
+            }
+        });
+
+        sendButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                curBuilding.addBadgeToUserById(chosenBadge, lastUserToAnswer);
```

Comment:
are we sure we want the last user? how do we know if he is the one that helped/was chosen by the user?

Reply:
Thought about this too..
Currently it's not the last user to comment that get the badge but the last neighbor commenting (ie not the request opener)
So it catches the case that the user opend the request says thanks to the neighbor helping before closing the request.

However It doesn't catch the case that another neighbor comments before she has the change to resolve it.
I can't see any "quick" fixes for this, any thoughts?

Reply:
one option is to show a list of neighbors that commented to choose from but that might be a bit tedious and users will dismiss it. 
maybe combine the list with giving a badge? show a list of neighbors and the click will already be the option to share a badge
what do you think?

Reply:
So instead of choosing a badge, choose the neighbor?

Reply:
something like "say thanks to who helped you" and give a random badge
or after choosing a neighbor and then a badge

Reply:
But if the list only has one user? (Which i assume would be the "normal" case)

Reply:
We could divide it to two use cases, but I am afraid this will add more work than scheduled

Reply:
I think it well be more elegant.
However this is fixing a corner case so lets add it to our todo list and prioritize this with our other tasks :)